Fix CI due to Numpy warning

We should fix the new warning raised by NumPy:
```
__________________ ERROR collecting tests/checkpoints_test.py __________________
tests/checkpoints_test.py:28: in <module>
    from flax import config, core, errors, io, struct
flax/io.py:42: in <module>
    from tensorflow.io import gfile  # type: ignore
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
.venv/lib/python3.12/site-packages/tensorflow/__init__.py:468: in <module>
    importlib.import_module("keras.src.optimizers")
/usr/lib/python3.12/importlib/__init__.py:90: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
.venv/lib/python3.12/site-packages/keras/__init__.py:7: in <module>
    from keras import _tf_keras as _tf_keras
.venv/lib/python3.12/site-packages/keras/_tf_keras/__init__.py:1: in <module>
    from keras._tf_keras import keras
.venv/lib/python3.12/site-packages/keras/_tf_keras/keras/__init__.py:16: in <module>
    from keras import export as export
.venv/lib/python3.12/site-packages/keras/export/__init__.py:7: in <module>
    from keras.src.export.saved_model import ExportArchive as ExportArchive
.venv/lib/python3.12/site-packages/keras/src/export/__init__.py:3: in <module>
    from keras.src.export.onnx import export_onnx
.venv/lib/python3.12/site-packages/keras/src/export/onnx.py:10: in <module>
    from keras.src.export.tf2onnx_lib import patch_tf2onnx
.venv/lib/python3.12/site-packages/keras/src/export/tf2onnx_lib.py:8: in <module>
    if not hasattr(np, "object"):
           ^^^^^^^^^^^^^^^^^^^^^
.venv/lib/python3.12/site-packages/numpy/__init__.py:773: in __getattr__
    warnings.warn(
E   FutureWarning: In the future `np.object` will be defined as the corresponding NumPy scalar.```
```

Source: https://github.com/google/flax/actions/runs/20151179703/job/58407944174

How to fix, let's make temporary patches where we import related code, for example:
```diff
# flax/io.py:42: in <module>
- from tensorflow.io import gfile  # type: ignore
+ with warnings.catch_warnings():
+   warnings.simplefilter("ignore")
+   from tensorflow.io import gfile  # type: ignore
```
and also for other places:
```python
# flax/metrics/tensorboard.py:23: in <module>
from tensorboard.plugins.hparams import api as hparams_api
```

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -62,6 +62,8 @@ testing = [
     "tensorflow_datasets",
     "tensorflow>=2.12.0; python_version<'3.13'", # to fix Numpy np.bool8 deprecation error
     "tensorflow>=2.20.0; python_version>='3.13'",
+    # Temporary fix for https://github.com/google/flax/issues/5143
+    "keras<3.13",
     "torch",
     "treescope>=0.1.1; python_version>='3.10'",
     "cloudpickle>=3.0.0",

